crates/code_generation/src/lib.rs

```diff
@@ -1,36 +1,67 @@
 use cyclers::generate_cyclers;
+use execution::{generate_replayer_struct, generate_run_function};
 use perception_databases::generate_perception_databases;
 use proc_macro2::TokenStream;
 use quote::quote;
-use run::generate_run_function;
 use source_analyzer::{cyclers::Cyclers, structs::Structs};
 use structs::generate_structs;
 
 mod accessor;
 pub mod cyclers;
+pub mod execution;
 pub mod perception_databases;
-pub mod run;
 pub mod structs;
 pub mod write_to_file;
 
-pub fn generate(cyclers: &Cyclers, structs: &Structs) -> TokenStream {
-    let generated_cyclers = generate_cyclers(cyclers);
-    let generated_run = generate_run_function(cyclers);
-    let generated_structs = generate_structs(structs);
+pub fn generate(cyclers: &Cyclers, structs: &Structs, mode: Execution) -> TokenStream {
+    let generated_cyclers = match mode {
+        Execution::None => Default::default(),
+        Execution::Run | Execution::Replay => {
+            let cyclers = generate_cyclers(cyclers, mode);
+            quote! {
+                mod cyclers {
+                    #cyclers
+                }
+            }
+        }
+    };
+    let generated_execution = match mode {
+        Execution::None => Default::default(),
```

Comment:
Currently not. Maybe in the future, if tools want to consume the generated structs only. In the long run, it was also planned that the behavior simulator uses the generated cycler completely. This would make `Execution::None` obsolete. But I think for now it is not much overhead to keep it.

Reply:
Okay